src/GHC/Util/View.hs

```diff
@@ -53,3 +55,7 @@ instance View' (Pat GhcPs) PApp_' where
   view' (fromPParen' -> LL _ (ConPatIn (L _ x) (InfixCon lhs rhs))) =
     PApp_' (occNameString . rdrNameOcc $ x) [lhs, rhs]
   view' _ = NoPApp_'
+
+-- A lambda with no guards and no where clauses
+pattern SimpleLambda :: [Pat GhcPs] -> LHsExpr GhcPs -> LHsExpr GhcPs
```

Comment:
I added a pattern synonym instead of following the `View` convention, because it's less boilerplate.
The `View`s here have no recursive calls of `view` so they seem to be equivalent to pattern synonyms.

Reply:
Ah, `PatternSynonyms` is restricted
It's been available since `7.8.1` so it shouldn't be an issue. Is there some other reason to restrict it?

Reply:
No reason to restrict it. Pattern synonym away!